mono/metadata/sre.c

```diff
@@ -2336,52 +2337,66 @@ mono_reflection_get_custom_attrs_blob_checked (MonoReflectionAssembly *assembly,
 	return result;
 }
 
-/**
- * reflection_setup_internal_class:
- * @tb: a TypeBuilder object
- * @error: set on error
- *
- * Creates a MonoClass that represents the TypeBuilder.
- * This is a trick that lets us simplify a lot of reflection code
- * (and will allow us to support Build and Run assemblies easier).
- *
- * Returns TRUE on success. On failure, returns FALSE and sets @error.
- */
 static gboolean
-reflection_setup_internal_class (MonoReflectionTypeBuilderHandle ref_tb, MonoError *error)
+reflection_setup_class_hierarchy (MonoReflectionTypeBuilderHandle ref_tb, GHashTable *unparented, MonoError *error)
```

Comment:
This doesn't use ref_tb anymore, AFAICT. And it doesn't allocate new handles to do its work, so it doesn't need handle function enter/return, either.